UnityProject/Assets/Scripts/Weapons/Projectiles/Behaviours/KeneticProjectileDamageInterity.cs

```diff
@@ -0,0 +1,57 @@
+﻿using UnityEngine;
+using Weapons.Projectiles.Behaviours;
+using Weapons;
+using ScriptableObjects.Gun;
+
+/// <summary>
+/// Damages integrity on collision
+/// only for kenetic weapons
+/// </summary>
+public class KeneticProjectileDamageInterity : MonoBehaviour, IOnShoot, IOnHit
+{
+	private GameObject shooter;
+	private Gun weapon;
+
+	[SerializeField] private DamageData damageData = null;
+
+	public void OnShoot(Vector2 direction, GameObject shooter, Gun weapon, BodyPartType targetZone = BodyPartType.Chest)
+	{
+		this.shooter = shooter;
+		this.weapon = weapon;
+	}
+
+	public bool OnHit(RaycastHit2D hit)
+	{
+		return TryDamage(hit);
+	}
+
+	private bool TryDamage(RaycastHit2D hit)
+	{
+		var newDamage = damageData.Damage;
+		var coll = hit.collider;
+		var integrity = coll.GetComponent<Integrity>();
+		if (integrity == null) return false;
+		// checks if its a high atmosphere 
+		
+
+			if (MatrixManager.AtPoint((Vector3Int)hit.point.To2Int(), true).MetaDataLayer.Get(hit.transform.localPosition.RoundToInt()).GasMix.Pressure <= 50)
```

Comment:
I'm debugging now, this evaluates to false in a pressurized room. This means I'm doing the full damage in a pressurized room, I think we need the opposite.